docs/src/main/asciidoc/_admin-guide/subsystem-configuration/JAXRS.adoc

```diff
@@ -0,0 +1,113 @@
+    </resteasy-language-mapping>
+    <resteasy-media-type-param-mapping>mt</resteasy-media-type-param-mapping>
+    <resteasy-providers>
+        <class>com.bluemonkey.reader</class>
+        <class>com.bluemonkey.writer</class>
+    </resteasy-providers>
+</subsystem>
+
+*Note.* The use of hyphens is a Wildfly convention. The hyphens are translated into periods before
+the parameters are passed into RESTEasy so that they conform to the RESTEasy parameter names.
+
+*Note.* 
+A parameter value is of one of five types:
+
+1. boolean
+2. integer
+3. string
+4. list
+5. map
+
+Parameters "resteasy.disable.providers" and "resteasy.providers" are lists of classes, using the 
+syntax shown in the example, and "resteasy-jndi-resources" is a list of JNDI names.
+"resteasy-language-mappings" and "resteasy-media-type-mappings" are maps, using the syntax
+shown in the example.
+
+Things look a little different in the Command Line Interface. For example,
+
+[source]
+[standalone@localhost:9990 subsystem=jaxrs] :list-add(name=resteasy-disable-providers, value=org.jboss.resteasy.plugin.providers.DefaultTextPlain)
+{"outcome" => "success"}
+[standalone@localhost:9990 subsystem=jaxrs] :list-add(name=resteasy-disable-providers, value=org.jboss.resteasy.plugin.providers.StringTextStar)
+{"outcome" => "success"}
+[standalone@localhost:9990 subsystem=jaxrs] :read-attribute(name=resteasy-disable-providers)
+{
+    "outcome" => "success",
+    "result" => [
+        "org.jboss.resteasy.plugin.providers.DefaultTextPlain",
+        "org.jboss.resteasy.plugin.providers.StringTextStar"
+    ]
+}
+
+or
+
+[source]
+[standalone@localhost:9990 subsystem=jaxrs] :map-put(name=resteasy-media-type-mappings, key=xml, value=application/xml)
+{"outcome" => "success"}
+[standalone@localhost:9990 subsystem=jaxrs] :map-put(name=resteasy-media-type-mappings, key=unusual, value=application/unusual)
+{"outcome" => "success"}
+[standalone@localhost:9990 subsystem=jaxrs] :read-attribute(name=resteasy-media-type-mappings)
+{
+    "outcome" => "success",
+    "result" => {
+        "xml" => "application/xml",
+        "unusual" => "application/unusual"
+    }
+}
+
+For a discussion of the various parameters, see the RESTEasy User Guide.
+
+*Note.* One important thing to understand is that these parameters are global. That is, they
+apply to all deployments.
+
+*Note.* Another important fact is that once a parameter is set, by way of the CLI interface, for example,
+that parameter and its value will be passed to subsequent deployments as web.xml context parameters. Existing 
+deployments are not affected.
+
+*Note.* Since there parameters are global, the classes referred in "resteasy.providers" and "resteasy.disable.providers"
+must be available to all deployments. In practice, then, they are meant to enable
+or disable RESTEasy providers. Note that they can be used in conjunction with
+"resteasy-use-builtin-providers" to tailor a set of available providers. 
```

Comment:
Effectively (if I am understanding properly) there are only two places a provider can happen currently:

- In the deployment, which we discourage since those are not available to all deployments
- "Built-in" - I don't know how resteasy-use-builtin-providers relates to that.

(Additionally they could be specified via a drop-in module, but for that we would need to be able to specify the module the class is to be loaded from)

Since I don't know what resteasy-use-builtin-providers I don't know if it makes sense to keep resteasy.providers or not.

I realise this comment is a bit muddled but hopefully it triggers something for you :)

Reply:
Hi @kabir. RESTEasy comes with some enabled built-in providers, and they can all be disabled by setting "resteasy-use-builtin-providers" to "false". There is a discussion on https://issues.redhat.com/browse/EAP7-634 about that. The point was that disabling one provider meant disabling all of them and then adding the rest back in using "resteasy.providers". That was the motivation for introducing "resteasy.disable.providers".

Reply:
Btw, I think adding a "module" key would be great, but for next time.